### PR TITLE
Remove dead code from GenerateConstructor.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs [hunk 1]
```diff
@@ -29,8 +29,6 @@ protected internal class State
             public IList<RefKind> ParameterRefKinds { get; private set; }
             public ImmutableArray<ITypeSymbol> ParameterTypes { get; private set; }
 
-            public IMethodSymbol DelegatedConstructorOpt { get; private set; }
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs]: Never written to...

**Comment by sharwell** [comment]
❓ What's the relationship between this and #20632?

**Comment by CyrusNajmabadi** [comment]
@sharwell Nothing afaict.

**Comment by CyrusNajmabadi** [comment]
Preemptively merging.

**Pull request merged**